README.md

```diff
@@ -1 +1,99 @@
-# csv-processor
\ No newline at end of file
+# @teamhive/csv-processor
+
+`csv-processor` is a package that simplifies reading and transforming CSV data into other formats through TypeDI services.
+The typical usecase for the package would be converting a CSV file into individual SQL insert statements.
+This package handles all of the file I/O as well as recognizing line breaks within single or double quote 
+
+### Usage
+To install the package, run:
+```sh
+npm install --save @teamhive/csv-processor
+```
+
+The package itself defines two main export `ProcessInput` and `ProcessOutput`.  Both classes are 
+typedi services and should be constructed as typedi dependencies.
+
+#### ProcessOuput
+This service opens up a file to write to and line-by-line writes the input provided by the `addLine(columns: string[], values: string[])` function.  To open up the output file, call 
+
+```Typescript
+processOutput.initialize(filePath: string, transform: TransformInput)
+```
+
+This will open the file and wait for calls to `addLine()`.
+
+Once all lines have been written to the output, `close()` should be called.
+
+#### TransformOutput
+`TransformOuput` is an interface for creating a transform that converts the input of `addLine` into a string that will be printed to the output file.
+```Typescript
+export interface TransformOutput {
+    transformOutput(columns: string[], vals: string[]): string;
+}
+```
+
+When using `ProcessOutput`, an implementation of `TransformOutput` must be passed to teh initialize function. A transform is required for output since the arrays must be formatted as a string.
```

Comment:
```suggestion
When using `ProcessOutput`, an implementation of `TransformOutput` must be passed to the initialize function. A transform is required for output since the arrays must be formatted as a string.
```